jetty-servlet/src/main/java/org/eclipse/jetty/servlet/ErrorPageErrorHandler.java

```diff
@@ -75,7 +75,19 @@ public String getErrorPage(HttpServletRequest request)
             if (errorPage != null)
                 matchedThrowable = exClass;
 
-            th = (th instanceof ServletException) ? ((ServletException)th).getRootCause() : null;
+            if (th instanceof ServletException)
+            {
+                th = ((ServletException)th).getRootCause();
+                if (th != null)
+                {
+                    request.setAttribute(Dispatcher.ERROR_EXCEPTION_TYPE, th.getClass());
```

Comment:
I don't think it is correct to do this here as we are walking the hierarchy.
I think the resetting of `Dispatcher.ERROR_EXCEPTION_TYPE` (and probably `Dispatcher.ERROR_EXCEPTION`) should be done below, only if we have an error page AND we are not in the compliance mode.